### PR TITLE
sql: update EXPLAIN related document

### PR DESCRIPTION
**Pull request by erjiaqing**
Related PR: https://github.com/pingcap/tidb/pull/10030

### CODE REVIEW

File: sql/understanding-the-query-execution-plan.md [hunk 1]
```diff
@@ -75,6 +75,13 @@ mysql> EXPLAIN SELECT count(*) FROM trips WHERE start_date BETWEEN '2017-07-01 0
 
 在添加完索引后的新执行计划中，使用 `IndexScan_24` 直接读取满足条件 `start_date BETWEEN '2017-07-01 00:00:00' AND '2017-07-01 23:59:59'` 的数据，可以看到，估算的要扫描的数据行数从之前的 `19117643.00` 降到了现在的 `8166.73`。在测试环境中显示，这个查询的执行时间从 50.41 秒降到了 0.00 秒！
 
+## <span id="explain-for-connection">`EXPLAIN FOR CONNECTION` </span>
+
+`EXPLAIN FOR CONNECTION`是用来获得一个链接中最后执行的查询的执行计划，它的输出格式与`EXPLAIN`完全一致。但我们的实现与MySQL的实现有些许不同，除了输出格式之外，还有下列不同：
+
+1. MySQL返回的是**正在执行**的查询计划，而TiDB返回的是**最后执行**的查询计划。也就是说，你还可以获得刚刚完成的查询的计划。
```
Comment (zz-jason) [inline sql/understanding-the-query-execution-plan.md]:
```suggestion
1. MySQL 返回的是**正在执行**的查询计划，而 TiDB 返回的是**最后执行**的查询计划。也就是说，你还可以获得刚刚完成的查询的计划。
```

中英文之间记得加空格

**Comment by zz-jason** [comment]
@lilin90 @eurekaka PTAL

File: sql/understanding-the-query-execution-plan.md [hunk 2]
```diff
@@ -75,6 +75,13 @@ mysql> EXPLAIN SELECT count(*) FROM trips WHERE start_date BETWEEN '2017-07-01 0
 
 在添加完索引后的新执行计划中，使用 `IndexScan_24` 直接读取满足条件 `start_date BETWEEN '2017-07-01 00:00:00' AND '2017-07-01 23:59:59'` 的数据，可以看到，估算的要扫描的数据行数从之前的 `19117643.00` 降到了现在的 `8166.73`。在测试环境中显示，这个查询的执行时间从 50.41 秒降到了 0.00 秒！
 
+## <span id="explain-for-connection">`EXPLAIN FOR CONNECTION` </span>
+
+`EXPLAIN FOR CONNECTION` 是用来获得一个链接中最后执行的查询的执行计划，它的输出格式与 `EXPLAIN` 完全一致。但我们的实现与MySQL的实现有些许不同，除了输出格式之外，还有下列不同：
```
Comment (eurekaka) [inline sql/understanding-the-query-execution-plan.md]:
```suggestion
`EXPLAIN FOR CONNECTION` 是用来获得一个连接中最后执行的查询的执行计划，它的输出格式与 `EXPLAIN` 完全一致。但我们的实现与 MySQL 的实现有些许不同，除了输出格式之外，还有下列不同：
```

File: sql/understanding-the-query-execution-plan.md [hunk 3]
```diff
@@ -75,6 +75,13 @@ mysql> EXPLAIN SELECT count(*) FROM trips WHERE start_date BETWEEN '2017-07-01 0
...
+
+1. MySQL 返回的是**正在执行**的查询计划，而 TiDB 返回的是**最后执行**的查询计划。也就是说，你还可以获得刚刚完成的查询的计划。
```
Comment (eurekaka) [inline sql/understanding-the-query-execution-plan.md]:
```suggestion
1. MySQL 返回的是**正在执行**的查询计划，而 TiDB 返回的是**最后执行**的查询计划。
```

File: sql/mysql-compatibility.md [hunk 4]
```diff
@@ -65,6 +65,10 @@ TiDB 实现自增 ID 的原理是每个 tidb-server 实例缓存一段 ID 值用
 
 Performance schema 表在 TiDB 中返回结果为空。TiDB 使用 [Prometheus 和 Grafana](https://pingcap.com/docs/op-guide/monitor/#use-prometheus-and-grafana) 来监测性能指标。
 
+### 查询计划
+
+TiDB 的查询计划（`EXPLAIN` / `EXPLAIN FOR`）输出格式与 MySQL 差别较大，同时 `EXPLAIN FOR` 的输出内容与权限设置与 MySQL 不一致，参见[理解 TiDB 执行计划](./understanding-the-query-execution-plan.md)。
```
Comment (lilin90) [inline sql/mysql-compatibility.md]:
```suggestion
TiDB 的查询计划（`EXPLAIN`/`EXPLAIN FOR`）输出格式与 MySQL 差别较大，同时 `EXPLAIN FOR` 的输出内容与权限设置与 MySQL 不一致，参见[理解 TiDB 执行计划](/sql/understanding-the-query-execution-plan.md)。
```

File: sql/understanding-the-query-execution-plan.md [hunk 5]
```diff
@@ -13,7 +13,7 @@ TiDB 优化器会根据当前数据表的实际情况来选择最优的执行计
 
 - `EXPLAIN` 可以和 `SELECT`，`DELETE` 语句一起使用；
 - 执行 `EXPLAIN`，TiDB 会返回被 `EXPLAIN` 的 SQL 语句经过优化器后的最终物理执行计划。也就是说，`EXPLAIN` 展示了 TiDB 执行该 SQL 语句的完整信息，比如以什么样的顺序，什么方式 JOIN 两个表，表达式树长什么样等等。详见 [`EXPLAIN` 输出格式](#explain-output-format)；
-- TiDB 目前还不支持 `EXPLAIN [options] FOR CONNECTION connection_id`，将在未来支持它，详见 [#4351](https://github.com/pingcap/tidb/issues/4351)；
+- TiDB 支持 `EXPLAIN [options] FOR CONNECTION connection_id`, 但与 MySQL 的 `EXPLAIN FOR` 有些许区别，请参见 [`EXPLAIN FOR CONNECTION`](#explain-for-connection)。
```
Comment (lilin90) [inline sql/understanding-the-query-execution-plan.md]:
```suggestion
- TiDB 支持 `EXPLAIN [options] FOR CONNECTION connection_id`，但与 MySQL 的 `EXPLAIN FOR` 有一些区别，请参见 [`EXPLAIN FOR CONNECTION`](#explain-for-connection)。
```

File: sql/understanding-the-query-execution-plan.md [hunk 6]
```diff
@@ -75,6 +75,13 @@ mysql> EXPLAIN SELECT count(*) FROM trips WHERE start_date BETWEEN '2017-07-01 0
 
 在添加完索引后的新执行计划中，使用 `IndexScan_24` 直接读取满足条件 `start_date BETWEEN '2017-07-01 00:00:00' AND '2017-07-01 23:59:59'` 的数据，可以看到，估算的要扫描的数据行数从之前的 `19117643.00` 降到了现在的 `8166.73`。在测试环境中显示，这个查询的执行时间从 50.41 秒降到了 0.00 秒！
 
+## <span id="explain-for-connection">`EXPLAIN FOR CONNECTION` </span>
```
Comment (lilin90) [inline sql/understanding-the-query-execution-plan.md]:
```suggestion
## `EXPLAIN FOR CONNECTION`
```

File: sql/understanding-the-query-execution-plan.md [hunk 7]
```diff
@@ -75,6 +75,13 @@ mysql> EXPLAIN SELECT count(*) FROM trips WHERE start_date BETWEEN '2017-07-01 0
...
+
+`EXPLAIN FOR CONNECTION` 是用来获得一个连接中最后执行的查询的执行计划，它的输出格式与 `EXPLAIN` 完全一致。但我们的实现与 MySQL 的实现有些许不同，除了输出格式之外，还有下列不同：
```
Comment (lilin90) [inline sql/understanding-the-query-execution-plan.md]:
```suggestion
`EXPLAIN FOR CONNECTION` 用于获得一个连接中最后执行的查询的执行计划，其输出格式与 `EXPLAIN` 完全一致。但 TiDB 中的实现与 MySQL 不同，除了输出格式之外，还有以下区别：
```

File: sql/understanding-the-query-execution-plan.md [hunk 8]
```diff
@@ -75,6 +75,13 @@ mysql> EXPLAIN SELECT count(*) FROM trips WHERE start_date BETWEEN '2017-07-01 0
...
+
+1. MySQL 返回的是**正在执行**的查询计划，而 TiDB 返回的是**最后执行**的查询计划。
+2. MySQL 的文档中指出，MySQL 要求你的登陆用户与被查询的连接相同，或者拥有 `PROCESS` 权限，而 TiDB 则要求你的登陆用户与被查询的连接相同，或者拥有 `SUPER` 权限。
```
Comment (lilin90) [inline sql/understanding-the-query-execution-plan.md]:
```suggestion
2. MySQL 的文档中指出，MySQL 要求登录用户与被查询的连接相同，或者拥有 `PROCESS` 权限，而 TiDB 则要求登录用户与被查询的连接相同，或者拥有 `SUPER` 权限。
```

File: sql/understanding-the-query-execution-plan.md [hunk 9]
```diff
@@ -75,6 +75,13 @@ mysql> EXPLAIN SELECT count(*) FROM trips WHERE start_date BETWEEN '2017-07-01 0
 
 在添加完索引后的新执行计划中，使用 `IndexScan_24` 直接读取满足条件 `start_date BETWEEN '2017-07-01 00:00:00' AND '2017-07-01 23:59:59'` 的数据，可以看到，估算的要扫描的数据行数从之前的 `19117643.00` 降到了现在的 `8166.73`。在测试环境中显示，这个查询的执行时间从 50.41 秒降到了 0.00 秒！
 
+## `EXPLAIN FOR CONNECTION`
+
+`EXPLAIN FOR CONNECTION` 用于获得一个连接中最后执行的查询的执行计划，其输出格式与 `EXPLAIN` 完全一致。但 TiDB 中的实现与 MySQL 不同，除了输出格式之外，还有以下区别：
+
+1. MySQL 返回的是**正在执行**的查询计划，而 TiDB 返回的是**最后执行**的查询计划。
```
Comment (lilin90) [inline sql/understanding-the-query-execution-plan.md]: The order is not important here, it'd be better to use a bulleted list. :)

```suggestion
- MySQL 返回的是**正在执行**的查询计划，而 TiDB 返回的是**最后执行**的查询计划。
```

File: sql/understanding-the-query-execution-plan.md [hunk 10]
```diff
@@ -75,6 +75,13 @@ mysql> EXPLAIN SELECT count(*) FROM trips WHERE start_date BETWEEN '2017-07-01 0
...
+2. MySQL 的文档中指出，MySQL 要求登录用户与被查询的连接相同，或者拥有 `PROCESS` 权限，而 TiDB 则要求登录用户与被查询的连接相同，或者拥有 `SUPER` 权限。
```
Comment (lilin90) [inline sql/understanding-the-query-execution-plan.md]:
```suggestion
- MySQL 的文档中指出，MySQL 要求登录用户与被查询的连接相同，或者拥有 `PROCESS` 权限，而 TiDB 则要求登录用户与被查询的连接相同，或者拥有 `SUPER` 权限。
```

**Pull request merged**